include/exadg/structure/spatial_discretization/operators/linear_operator.cpp

```diff
@@ -35,11 +35,14 @@ LinearOperator<dim, Number>::do_cell_integral(IntegratorCell & integrator) const
 
   for(unsigned int q = 0; q < integrator.n_q_points; ++q)
   {
-    // engineering strains (material tensor is symmetric)
-    tensor const gradient = integrator.get_gradient(q);
+    // engineering strains
+    tensor const symmetric_gradient = integrator.get_symmetric_gradient(q);
 
-    // Cauchy stresses
-    tensor const sigma = material->apply_C(gradient, integrator.get_current_cell_index(), q);
+    // Cauchy stresses, only valid for linear elasticity
+    tensor const sigma =
+      material->second_piola_kirchhoff_stress(symmetric_gradient,
```

Comment:
This PR replaces `apply_C()` by `second_piola_kirchhoff_stress_derivative()`, so I would expect that `second_piola_kirchhoff_stress_derivative()` is called here. The new version should also be valid, but I think not all of these changes should be mixed into a single PR.

Reply:
I think actually in the `linear_operator.cpp` calling the derivative is more confusing, as we only want to evaluate the stress actually! ...once again one of these points were linear elasticity and St Venant solid were very similar, but other more general models are not so much as far as I am concerned.